sphinxext/opengraph/__init__.py

```diff
@@ -28,6 +31,45 @@
 }
 
 
+def image_abs_url(image_path: str, url: str, docname: str = None, app: Sphinx = None):
+    parsed_url = urlparse(image_path)
+
+    # If image_uri is None or if it is an url (contains a scheme) leave it unchanged
+    if not image_path or parsed_url.scheme:
+        return image_path
+
+    if docname and app:
+        # Convert local image path to an absolute url and make sure image gets copied on build
+        return urljoin(url, note_image(parsed_url.path, docname, app))
+    else:
+        # If the app and docname aren't specified, assume that the path is a path relative to the output file
+        return urljoin(url, image_path)
+
+
+def note_image(image_path: str, docname: str, app: Sphinx):
```

Comment:
Open for comments on any of the matters mentioned in the comments in this function